### PR TITLE
Create patching-troubleshooting.md

### PR DESCRIPTION
**Pull request by gnivler**
Just thought I would try hammering something out that brings together common questions, FAQ of sorts?  No offense taken if not desired

### CODE REVIEW

**Comment by pardeike** [comment]
We have to make sure it does not overlap with https://harmony.pardeike.net/articles/patching-edgecases.html (or refactor that).

File: Harmony/Documentation/articles/patching-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+## Troubleshooting
+
+### Dll
+
+- make sure you're actually running the built DLL you expect you are (post-build copy problems)
+
+### Log
+
+- make sure your patch is being run by logging something out, from a Postfix or Finalizer
```
Comment (pardeike) [inline Harmony/Documentation/articles/patching-troubleshooting.md:9]: Add: manual patching / TargetMethod - add NULL checks, logging

File: Harmony/Documentation/articles/patching-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+## Troubleshooting
+
+### Dll
+
+- make sure you're actually running the built DLL you expect you are (post-build copy problems)
```
Comment (pardeike) [inline Harmony/Documentation/articles/patching-troubleshooting.md:5]: wording: more like "that the harmony dll you compile against is already loaded" or so

File: Harmony/Documentation/articles/patching-troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+### Log
+
+- make sure your patch is being run by logging something out, from a Postfix or Finalizer
+
+### Problems finding methods and other members
+
+- specify signature arguments
```
Comment (pardeike) [inline Harmony/Documentation/articles/patching-troubleshooting.md:13]: Add: ref/out types

File: Harmony/Documentation/articles/patching-troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+- enumerate and select with reflection (use AccessTools https://harmony.pardeike.net/api/HarmonyLib.AccessTools.html)
```
Comment (pardeike) [inline Harmony/Documentation/articles/patching-troubleshooting.md:14]: Add: example

File: Harmony/Documentation/articles/patching-troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+- nested class (wrong hierarchical level)
+
+### Something isn't changing
+
+- do you need `ref` on your argument?
```
Comment (pardeike) [inline Harmony/Documentation/articles/patching-troubleshooting.md:19]: Add: note/link to inlining

**Comment by pardeike** [comment]
Is anything still happening on this? As it is, its not very useful.

**Comment by gnivler** [comment]
Sorry for the lack of reply, I had email forwarding problems.  No worries
about the content.

**Pull request closed without merging**